master/buildbot/db/logs.py

```diff
@@ -356,11 +356,21 @@ def thddeleteOldLogs(conn):
 
             # update log types older than timestamps
             # we do it first to avoid having UI discrepancy
+
+            # SELECT steps.id from steps WHERE steps.started_at < older_than_timestamp ORDER BY steps.id DESC LIMIT 1;
+            res = conn.execute(
+                sa.select([model.steps.c.id])
+                .where(model.steps.c.started_at < older_than_timestamp)
+                .order_by(model.steps.c.id.desc())
+                .limit(1)
+            )
+            stepid_max = res.fetchone()[0]
```

Comment:
Could we add a comment explaining this? Something like: "we utilize the fact that steps.id is autoincrement, thus step start times are effectively sorted and we only need to find the step at the upper bound of steps to update.

Reply:
Will do :)